client/purplecaffeine/widget.py

```diff
@@ -0,0 +1,337 @@
+            pane_widths=[1, 0, 5],
+        )
+
+    def render_table(self, name_value_list):
+        """
+        Method to construct a new html string representing
+        a table using name and values.
+
+        Returns:
+            table string (str): html string that contains the table
+        """
+        if len(name_value_list) == 0:
+            return ""
+        parameter_rows = "".join(
+            [
+                f"<tr><td style='text-align: center;'>{str(name)}</td>"
+                + "<td style='text-align: center;'><button class='btn btn-primary "
+                f"rounded-pill' disabled>{str(value)}</button"
+                f"></td></tr>"
+                for name, value in name_value_list
+            ]
+        )
+        return (
+            "<table border=2 class='table'><tr>"
+            "<th style='text-align: center;'>Parameter</th>"
+            "<th style='text-align: center;'>Value</th></tr>"
+            f"{parameter_rows}</table>"
+        )
+
+    def render_trial(self):
+        """
+        Load the tabs with the basic info, circuits and metrics
+        of the selected trial
+        """
+        if self.selected_trial is None:
+            return self.display_message("Add a new trial to see the info of that trial")
+
+        tags = "".join(
+            [
+                f"<button class='btn btn-primary rounded-pill btn-sm' "
+                f"style='margin-bottom:20px; margin-right: 10px' "
+                f"disabled>{str(tag)}</button>"
+                for tag in self.selected_trial.tags
+            ]
+        )
+
+        description = (
+            "<p>Description: " + f"{self.selected_trial.description}</p>"
+            if self.selected_trial.description
+            else ""
+        )
+
+        info = widgets.HTML(
+            '<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/'
+            'bootstrap@5.2.3/dist/css/bootstrap.min.css" '
+            'integrity="sha384-rbsA2VBKQhggwzxH7pPCaAqO46MgnOM80zW1RWuH61DGLwZJEdK2Kadq2F9CUG65" '
+            'crossorigin="anonymous">'
+            f"<h3>{self.selected_trial.name} | {self.selected_trial.uuid} </h3>"
+            + description
+            + f"<div>{tags}</div>"
+            + self.render_table(self.selected_trial.parameters)
+        )
+        info.layout = Layout(overflow="scroll", max_height="300px")
+
+        metrics = widgets.HTML(
+            "<h3 style='text-align:center'> Metrics </h3>"
+            + (self.render_table(self.selected_trial.metrics))
+        )
+
+        metrics.layout = Layout(overflow="scroll", max_height="300px")
+        circuits = widgets.Output()
+        with circuits:
+            for name, circuit in self.selected_trial.circuits:
+                print(name)
+                print(circuit)
+        circuits.layout = Layout(overflow="scroll", max_height="300px")
+
+        tab = widgets.Tab(children=[info, metrics, circuits])
+        tab.titles = ["Info", "Metrics", "Circuits"]
```

Comment:
Why do you only have 3 tabs ?
Trial has :
- metrics
- parameters
- circuits
- operators
- artifacts
- arrays

Reply:
this is MVP info will have parameters, title, description; metrics -> metrics, circuit -> circuits. Operators, artifacts, texts and arrays will be added later